src/audio/pipeline/pipeline-schedule.c

```diff
@@ -53,7 +53,16 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 		switch (cmd) {
 		case COMP_TRIGGER_STOP:
 		case COMP_TRIGGER_PAUSE:
-			if (p->trigger.aborted)
+			/* if the trigger was aborted, keep the task running */
+			if (p->trigger.aborted) {
+				ret = SOF_TASK_STATE_RUNNING;
+				break;
+			}
+
+			/*
+			 * if there's a trigger pending, keep the task running until it is executed
+			 */
+			if (p->trigger.pending)
 				ret = SOF_TASK_STATE_RUNNING;
 			else
 				ret = SOF_TASK_STATE_COMPLETED;
```

Comment:
(to the second commit, github seems to show this already with the first one) so this all means
```
	if (p->trigger.pending || p->trigger.aborted)
		ret = SOF_TASK_STATE_RUNNING;
	else
		ret = SOF_TASK_STATE_COMPLETED;
	break;
```
Shouldn't we reset `.pending` too immediately after processing it?